### ISSUE TITLE
Regression on `read_json_auto` performance between 0.7 and 0.8 for gzipped files

### ISSUE THREAD

**Issue by b-per** [duckdb/duckdb#7729]
### What happens?

I am reading some nested JSON and found some discrepancies between version 0.7 (which used to work fine), and 0.8

Here are different outputs for duckdb 0.7 (named `duckdb7`) and 0.8 (named `duckdb`) 

```
# with 3000 entries in the sample

time duckdb7 :memory: "select *  from read_json_auto('myfile.json.gz', sample_size=3000) limit 0;"
duckdb7 :memory:   0.18s user 0.01s system 100% cpu 0.196 total

time duckdb :memory: "select *  from read_json_auto('myfile.json.gz', sample_size=3000) limit 0;"
duckdb :memory:   0.32s user 0.04s system 100% cpu 0.357 total

# with 5000 entries in the sample

time duckdb7 :memory: "select *  from read_json_auto('myfile.json.gz', sample_size=5000) limit 0;"
duckdb7 :memory:   0.27s user 0.02s system 99% cpu 0.297 total

time duckdb :memory: "select *  from read_json_auto('myfile.json.gz', sample_size=5000) limit 0;"
# does not finish after several minutes, the CPU is at 100% and Memory usage is very low (2%)
```

Adding `maximum_depth=1` did not change the 0.8 behaviour

I tried the non-gzipped version of my file:
- duckdb 0.7 returned results in 0.097s
- duckdb 0.8 successfully returned results this time, but in 0.398s (x4 compared to 0.7)

### To Reproduce

Run `read_json_auto` on a gzipped JSON file.

### OS:

x86_64

### DuckDB Version:

v0.8.0 e8e4cea5ec

### DuckDB Client:

CLI

### Full Name:

Benoit Perigaud

### Affiliation:

NA

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by b-per** [comment]
Just reporting back after the fix, running on `v0.8.1-dev287 8c32403411`

`read_json_auto()` now successfully returns results even on a `json.gz` file. It is about 3x slower than v0.7 though

```
time duckdb7 :memory: "select *  from read_json_auto('myfile.json.gz') limit 0;"
duckdb7 :memory:   0.17s user 0.02s system 99% cpu 0.195 total

time duckdb :memory: "select *  from read_json_auto('myfile.json.gz') limit 0;"
duckdb :memory:   0.69s user 0.07s system 100% cpu 0.757 total
```

**Comment by lnkuiper** [comment]
@b-per With 0.8.0, I've increased the block size when reading JSON because our users tended to have very large JSON objects, which might limit parallelism in your case. So we've sacrificed some performance in favor of being more user-friendly.
